templates/guest_book/gulpfilelocal.js

```diff
@@ -0,0 +1,120 @@
+const gulp = require("gulp");
+const fs = require('fs');
+const args = require("yargs").argv;
+const { Account, SimpleKeyStoreSigner, InMemoryKeyStore, KeyPair, LocalNodeConnection, NearClient, Near } = require('nearlib');
+
+gulp.task('copyfiles', function () {
+  return gulp.src('./assembly/**/*')
+    .pipe(gulp.dest('./out/'));
+});
+
+gulp.task("build:model", callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "model.ts",
+    "--baseDir", "./out",
+    "--nearFile", "../out/model.near.ts",
+    "--measure"
+  ], callback);
+});
+
+gulp.task("build:bindings", ["build:model"], callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "main.ts",
+    "--baseDir", "./out",
+    "--binaryFile", "../out/main.wasm",
+    "--nearFile", "../out/main.near.ts",
+    "--measure"
+  ], callback);
+});
+
+gulp.task("build:all", ["build:bindings"], callback => {
+  const asc = require("assemblyscript/bin/asc");
+  asc.main([
+    "../out/main.near.ts",
+    "--baseDir", "./out",
+    "-O3",
+    "--binaryFile", "../out/main.wasm",
+    "--sourceMap",
+    "--measure"
+  ], callback);
+});
+
+gulp.task('build', function(done) {
+  const runSequence = require('run-sequence');
+  runSequence('copyfiles', 'build:all', function() {
+    done();
+  });
+});
+
+gulp.task('deploy', function(done) {
+  // Find keys/ account id
+  const keyPath = './neardev/devkey.json';
+  if (!fs.existsSync(keyPath)) {
+    console.error("Deployment failed. File ./neardev/devkey.json does not exist. Please add this file and try again. Please make sure file contains public_key, secret_key, and account_id.")
+    return;
+  }
+  const rawKey = JSON.parse(fs.readFileSync(keyPath));
+  if (!rawKey.public_key || !rawKey.secret_key || !rawKey.account_id) {
+    console.error("Deployment failed. neardev/devkey.json format problem. Please make sure file contains public_key, secret_key, and account_id")
+    return;
+  }
+  const keyPair = new KeyPair(rawKey.public_key, rawKey.secret_key);
+  const accountId = rawKey.account_id;
+  const keyStore = new InMemoryKeyStore();
+  keyStore.setKey(accountId, keyPair);
+
+  // Determine which url to use
+  const nodeUrl = getNodeUrl(args);
+  const localNodeConnection = new LocalNodeConnection(nodeUrl);
+  const nearClient = new NearClient(new SimpleKeyStoreSigner(keyStore), localNodeConnection);
+  const near = new Near(nearClient);
+  const contractData = [...fs.readFileSync('./out/main.wasm')];
+
+  // Contract name
+  const contractName = getContractName(args);
+  console.log(
+    "Starting deployment. Account id " + accountId + ", contract " + contractName + ", url " + nodeUrl);
+  deployContractAndWaitForTransaction(
+    accountId, contractName, contractData, near)
+    .then(
```

Comment:
https://gulpjs.com/docs/en/getting-started/async-completion#returning-a-promise it's ok to return a promise from a gulp task, so you can write with `async` / `await` style for simpler logic

Reply:
done